src/phases/obtain-status-effect-phase.ts

```diff
@@ -44,6 +46,12 @@ export class ObtainStatusEffectPhase extends PokemonPhase {
               this.sourceText ?? undefined,
             ),
           );
+          if (this.statusEffect && this.statusEffect !== StatusEffect.FAINT) {
```

Comment:
```suggestion
          if (!isNullOrUndefined(this.statusEffect) && this.statusEffect !== StatusEffect.FAINT) {
```
Also needs imports updated.